### PR TITLE
Refactor middleware

### PR DESCRIPTION
**Pull request by shogo82148**
## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling functionality for improved clarity and type safety.
	- Introduced a new type for managing request IDs in logging.
	- Implemented a fully functional test suite for the logging middleware.
	- Added comprehensive tests for Request ID functionality, ensuring edge cases are addressed.
	- Streamlined logging response tests to improve clarity and maintainability.

- **Bug Fixes**
	- Refactored log context handling to prevent potential runtime errors.

- **Refactor**
	- Updated data types in various middleware components to utilize modern Go conventions (e.g., `any` instead of `interface{}`).
	- Transitioned to standard Go testing practices, simplifying test code.
	- Improved control flow and readability in the header validation logic.
	- Restructured tests to enhance readability and maintainability.
	- Renamed and reorganized timeout tests for improved clarity and functionality.

### CODE REVIEW

File: middleware/log_response_test.go [hunk 1]
```diff
@@ -1,50 +1,52 @@
...
-// 		req, err = http.NewRequest("POST", "/goo", strings.NewReader(`{"payload":42}`))
-// 		Ω(err).ShouldNot(HaveOccurred())
-// 		rw = new(testResponseWriter)
-// 		params = url.Values{"query": []string{"value"}}
-// 		ctx = newContext(service, rw, req, params)
-// 	})
+	// call the handler
+	handler := func(ctx context.Context, rw http.ResponseWriter, req *http.Request) error {
+		shogoa.ContextResponse(ctx).WriteHeader(200)
+		shogoa.ContextResponse(ctx).Write([]byte("some response data to be logged"))
```
[the remark this answers is not in the document]
Comment (shogo82148) [reply middleware/log_response_test.go:35]: nice catch!

[the remark this answers is not in the document]
Comment (shogo82148) [reply middleware/log_response_test.go]: fixed in https://github.com/shogo82148/shogoa/pull/33/commits/02e2c30b9a8c91abbdc6be736268a253cf7720dd
